Sources/tart/Commands/Push.swift

```diff
@@ -54,48 +54,43 @@ struct Push: AsyncParsableCommand {
         + "\(registryIdentifier.host)/\(registryIdentifier.namespace)\(remoteNamesForRegistry.referenceNames())...")
 
       let references = remoteNamesForRegistry.map{ $0.reference.value }
-      let pushedRemoteName: RemoteName
-      let lightweightPush: Bool
 
+      let pushedRemoteName: RemoteName
       // If we're pushing a local OCI VM, check if points to an already existing registry manifest
       // and if so, only upload manifests (without config, disk and NVRAM) to the user-specified references
-      if let remoteName = try? RemoteName(localName),
-         let remoteName = try await lightweightPushToRegistry(registry: registry, remoteName: remoteName,
-                                                              references: references) {
-        pushedRemoteName = remoteName
-        lightweightPush = true
+      if let remoteName = try? RemoteName(localName) {
+        pushedRemoteName = try await lightweightPushToRegistry(
+          registry: registry, 
+          remoteName: remoteName,
+          references: references
+        )
       } else {
         pushedRemoteName = try await localVMDir.pushToRegistry(
           registry: registry,
           references: references,
           chunkSizeMb: chunkSize
         )
-        lightweightPush = false
-      }
-
-      // Populate the local cache (if requested)
-      if populateCache {
-        let ociStorage = VMStorageOCI()
-
-        if !lightweightPush {
+        // Populate the local cache (if requested)
+        if populateCache {
           let expectedPushedVMDir = try ociStorage.create(pushedRemoteName)
           try localVMDir.clone(to: expectedPushedVMDir, generateMAC: false)
         }
+      }
 
+      // link the rest remote names
+      if populateCache {
         for remoteName in remoteNamesForRegistry {
           try ociStorage.link(from: remoteName, to: pushedRemoteName)
         }
       }
     }
   }
 
-  func lightweightPushToRegistry(registry: Registry, remoteName: RemoteName, references: [String]) async throws -> RemoteName? {
+  func lightweightPushToRegistry(registry: Registry, remoteName: RemoteName, references: [String]) async throws -> RemoteName {
     // Is the local OCI VM already present in the registry?
     let digest = try VMStorageOCI().digest(remoteName)
 
-    guard let (remoteManifest, _) = try? await registry.pullManifest(reference: digest) else {
-      return nil
-    }
+    let (remoteManifest, _) = try await registry.pullManifest(reference: digest)
```

Comment:
This will throw an exception if the remote reference does not exist.

This is not desirable, because we want to be opportunistic here: try a lightweight path first and then fallback to a heavyweight one if the former fails.

Reply:
See my previous comment. I think it's reasonable to fail here.